.golangci.yml

```diff
@@ -0,0 +1,153 @@
+# options for analysis running
+run:
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 1m
+
+output:
+  sort-results: true
+
+# Uncomment and add a path if needed to exclude
+# skip-dirs:
+#   - some/path
+# skip-files:
+#   - ".*\\.my\\.go$"
+#   - lib/bad.go
+
+# Find the whole list here https://golangci-lint.run/usage/linters/
+linters:
+  disable-all: true
+  enable:
+    - deadcode # finds unused code
+    - errcheck # checking for unchecked errors in go programs
+    - errorlint # errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.
+    # - goconst # finds repeated strings that could be replaced by a constant
+    - dupl # tool for code clone detection
+    - forbidigo # forbids identifiers	by reg exps
+    - gomoddirectives # manage the use of 'replace', 'retract', and 'excludes' directives in go.mod.
+    - gosimple # linter for Go source code that specializes in simplifying a code
+    - misspell # finds commonly misspelled English words in comments
+    - nakedret # finds naked returns in functions greater than a specified function length
+    - prealloc # finds slice declarations that could potentially be preallocated
+    - nolintlint # reports ill-formed or insufficient nolint directives
+    - staticcheck # Staticcheck is a go vet on steroids, applying a ton of static analysis checks
+    - stylecheck # a replacement for golint
+    - unparam # reports unused function parameters
+    - unused # checks Go code for unused constants, variables, functions and types
+
+    - govet # Vet examines Go source code and reports suspicious constructs, such as Printf calls whose arguments do not align with the format string
+    - ineffassign # detects when assignments to existing variables are not used
+    - structcheck # finds unused struct fields
+    - typecheck # Like the front-end of a Go compiler, parses and type-checks Go code
+    - varcheck # Finds unused global variables and constants
+    - asciicheck # simple linter to check that your code does not contain non-ASCII identifiers
+    - bodyclose # checks whether HTTP response body is closed successfully
+    - durationcheck # check for two durations multiplied together
+    - exportloopref # checks for pointers to enclosing loop variables
+    - forcetypeassert # finds forced type assertions
+    - goimports # Goimports does everything that gofmt does. Additionally it checks unused imports
```

Comment:
@belimawr what you're suggesting is more for an editor, is not it?

This just checks that the imports are used and tells you if there is something wrong, it does not change your code. Although the linter has the `fix: true` configuration parameters, I think we should fix all the linting issues manually to understand them better and to avoid repeating.

Reply:
I'm just concerned that using `gci` as a linter would force every single member of our team to run it in their editor, otherwise it would not satisfy the check.

Reply:
@belimawr I tried to run the code base against `gci` and I get `File is not `gci`-ed (gci)` for most of the files where `goimports` passes just fine. I'm not in favour, I don't think grouping imports is so important.

Reply:
> @belimawr I tried to run the code base against `gci` and I get `File is not `gci`-ed (gci)` for most of the files where `goimports` passes just fine. I'm not in favour, I don't think grouping imports is so important.

I haven't tried it on Beats, in my previous job the projects were small and we were adding imports all the time, so it helped to keep things nice and tidy. Otherwise we'd  end up with our imports split into 20 different groups with no meaning.

> I'm just concerned that using gci as a linter would force every single member of our team to run it in their editor

That's not quite true because it could be part of `mage fmt` which we already have to run, so it'd be mostly transparent.

Anyway, that's far from a must-have to me. If I'm the only one who likes it, I'm happy to go without it ;)

Reply:
For what it is worth, I also like grouping imports, but not enough to force it on people. 

Import grouping isn't the most important problem in the code the linter will find. We could look at enabling this later after we have the linter enabled everywhere (if we find it is annoying).